"Jest Test Current eslint-plugin Rule" shouldn't block on TS checker errors

**Repro**

1. `code .` in the root repository directory on a Windows 10 computer
2. Open an eslint-plugin rule
3. Add an unused variable
4. Press F5 to launch `Jest Test Current eslint-plugin Rule`

**Expected Result**

That rule's `.test.ts` file should be debugged.

**Actual Result**

```
Debugger attached.
 FAIL  tests/rules/base-to-string.test.ts
  ● Test suite failed to run

    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/rules/base-to-string.ts:36:19 - error TS6133: 'type' is declared but its value is never read.

    36             const type = typeChecker.getTypeAtLocation(parserServices.esTreeNodeToTSNodeMap.get(node));
                         ~~~~
```

**Additional Info**

It's not unheard of that we'd want to add unused variables during testing. Can we remove these extra checks? 

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | N/A |
| `@typescript-eslint/parser`        | N/A |
| `TypeScript`                       | N/A |
| `ESLint`                           | N/A |
| `node`                             | `12.0.0` |
| `npm`                              | `6.9.0` |

--- a/packages/eslint-plugin-internal/jest.config.js
+++ b/packages/eslint-plugin-internal/jest.config.js
@@ -1,6 +1,11 @@
 'use strict';
 
 module.exports = {
+  globals: {
+    'ts-jest': {
+      isolatedModules: true,
+    },
+  },
   testEnvironment: 'node',
   transform: {
     '^.+\\.tsx?$': 'ts-jest',

--- a/packages/eslint-plugin-tslint/jest.config.js
+++ b/packages/eslint-plugin-tslint/jest.config.js
@@ -1,6 +1,11 @@
 'use strict';
 
 module.exports = {
+  globals: {
+    'ts-jest': {
+      isolatedModules: true,
+    },
+  },
   testEnvironment: 'node',
   transform: {
     '^.+\\.tsx?$': 'ts-jest',

--- a/packages/eslint-plugin/jest.config.js
+++ b/packages/eslint-plugin/jest.config.js
@@ -1,6 +1,11 @@
 'use strict';
 
 module.exports = {
+  globals: {
+    'ts-jest': {
+      isolatedModules: true,
+    },
+  },
   testEnvironment: 'node',
   transform: {
     '^.+\\.tsx?$': 'ts-jest',

--- a/packages/experimental-utils/jest.config.js
+++ b/packages/experimental-utils/jest.config.js
@@ -1,6 +1,11 @@
 'use strict';
 
 module.exports = {
+  globals: {
+    'ts-jest': {
+      isolatedModules: true,
+    },
+  },
   testEnvironment: 'node',
   transform: {
     '^.+\\.tsx?$': 'ts-jest',

--- a/packages/parser/jest.config.js
+++ b/packages/parser/jest.config.js
@@ -1,6 +1,11 @@
 'use strict';
 
 module.exports = {
+  globals: {
+    'ts-jest': {
+      isolatedModules: true,
+    },
+  },
   testEnvironment: 'node',
   transform: {
     '^.+\\.tsx?$': 'ts-jest',

--- a/packages/typescript-estree/jest.config.js
+++ b/packages/typescript-estree/jest.config.js
@@ -1,6 +1,11 @@
 'use strict';
 
 module.exports = {
+  globals: {
+    'ts-jest': {
+      isolatedModules: true,
+    },
+  },
   testEnvironment: 'node',
   transform: {
     '^.+\\.tsx?$': 'ts-jest',

